### PR TITLE
[moved] | Update about page

### LINKED ISSUE #1700: Update Scratch About Page - Credits
Page: https://scratch.mit.edu/about/
Section: "Support and Funding"

Request:
Update list to include Siegel Family Endowment

Current copy:
The Scratch project has received financial support from the National Science Foundation, Scratch Foundation, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, and MacArthur Foundation.

New copy:
The Scratch project has received financial support from the National Science Foundation, Scratch Foundation, Siegel Family Endowment, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, and MacArthur Foundation.

### PR DESCRIPTION
**Pull request by VutonDesign**
#### Info
Updates the "Support and Funding" section of the about page.

#### Fixes
#1700

### CODE REVIEW

**Comment by chrisgarrity** [comment]
@VutonDesign I confirmed with Champika that it's ok to make the change to the copy to make the supporters a list without the word 'and'. See the issue for the new copy.

**Comment by VutonDesign** [comment]
Ah, I'm not too familiar with FormattedHTMLMessage or FormatMessage so I don't know if I would be able to make that change. :thinking:

**Comment by VutonDesign** [comment]
Wait, I may be able to do most of it...

To check I know what I'm doing, would an example of the support and funding section in FormatMessage be:
```
"about.support": "Support and Funding",
"about.supportDescription": "The Scratch project has received financial support from the following organizations: National Science Foundation, Scratch Foundation, Siegel Family Endowment, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, MacArthur Foundation. See the {creditslink} for more information. If you&#39;d like to support Scratch, please see the Scratch Foundation {donatelink}, or contact us at {donateemail}"
```
```
<FormattedMessage 
    id='about.supportDescription'
    values={{
        creditslink: (
            <a href='//scratch.mit.edu/info/credits'>
                credits page
            </a>
        )
    }}
/>
etc.

**Comment by chrisgarrity** [comment]
@VutonDesign yes, that's the general approach. 

A couple of things in your example...
- The supporterList should get added as a placeholder. 
- The link text also needs to be translated. 
So you need to add new strings to the l10n.json like:
```
"about.creditsLinkText": "credits page",
"about.donateLinkText": "donate page",
"about.supportDescription": "The Scratch project has received financial support from the following organizations: {supporterList}. See the {creditslink} for more information. If you&#39;d like to support Scratch, please see the Scratch Foundation {donatelink}, or contact us at {donateemail}"
```
and then the values become:
```
    values={{
        supporterLink: 'National Science Foundation, Scratch Foundation, Siegel Family Endowment, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, MacArthur Foundation',
        creditslink: (
            <a href='//scratch.mit.edu/info/credits'>
                <FormattedMessage id='about.creditLinkText'/>
            </a>
        ),
        donatelink: (
            <a href='//secure.donationpay.org/scratchfoundation/'>
                <FormattedMessage id='about.donateLinkText'/>
            </a>
        ),
        donateemail: (
            <a href='mailto:donate@scratch.mit.edu'>
                donate@scratch.mit.edu
            </a>
        )
    }}

```

The supporterList and donate email strings don't need to be translated, so they don't need to be added to the l10n.

**Comment by VutonDesign** [comment]
Okay, I'll give it a go 👍

**Comment by VutonDesign** [comment]
@chrisgarrity I'll need to leave the rest of this for tomorrow, I think it's mostly just indentations issues that need to be fixed.

File: src/views/about/about.jsx [hunk 1]
```diff
@@ -60,6 +60,14 @@ var About = React.createClass({
                             <h3><FormattedMessage id='about.aroundTheWorld' /></h3>
                             <img src="/images/about/around-the-world.png" alt="" />
                             <p><FormattedHTMLMessage id='about.aroundTheWorldDescription' /></p>
+                            values={{
```
Comment (chrisgarrity) [inline src/views/about/about.jsx]: This should be FormattedMessage (not FormattedHTMLMessage) 
The placeholder in `'about.aroundTheWorldDescription'` is `{translationLink}`, so the values object just needs to specify the value for `translationLink`
```jsx
values={{
    translationLink: (
        <a href='http://wiki.scratch.mit.edu/wiki/How_to_Translate_Scratch\'>
            <FormattedMessage id='about.translationLinkText' />
        </a>
    )
}}
```
That is, the name in the placeholder becomes the key in the values object, and the value is the link along with the translation of the link text.

File: src/views/about/about.jsx [hunk 2]
```diff
@@ -72,12 +80,36 @@ var About = React.createClass({
                             <h3><FormattedMessage id='about.quotes' /></h3>
                             <img src="/images/about/quotes.gif" alt="Quotes about Scratch" />
                             <p><FormattedHTMLMessage id='about.quotesDescription' /></p>
+                            values={{
```
Comment (chrisgarrity) [inline src/views/about/about.jsx]: remember to replace the FormattedHTMLMessage with just FormattedMessage.

Comment (chrisgarrity) [reply src/views/about/about.jsx]: I think you'll need to go over all the FormattedHTMLMessages, changing them to `FormattedMessage` and double checking the values against the placeholder names.

Thanks for noticing and updating the other strings in the view! When you're all done you should be able to remove the `require(require('react-intl').FormattedHTMLMessage` on line 2 as well.

File: src/views/about/about.jsx (hunk 1 above)
Comment (VutonDesign) [reply src/views/about/about.jsx]: Okay 👍

File: src/views/about/about.jsx (hunk 2 above)
Comment (VutonDesign) [reply src/views/about/about.jsx]: 👍

**Comment by VutonDesign** [comment]
The Travis build came back with this:
```  138:1   error  Line 138 exceeds the maximum line length of 120           max-len```

How would I fix this? (line 138 is the supporters list)

File: src/views/about/about.jsx [hunk 3]
```diff
@@ -102,8 +133,26 @@ var About = React.createClass({
 
                         <li>
                             <h3><FormattedMessage id='about.support' /></h3>
-                            <p><FormattedHTMLMessage id='about.supportDescription' /></p>
-                        </li>
+                            <p><FormattedMessage id='about.supportDescription' /></p>
+                            values={{
+        supportersList: 'National Science Foundation, Scratch Foundation, Siegel Family Endowment, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, MacArthur Foundation',
```
Comment (Kenny2github) [inline src/views/about/about.jsx]: Shorten this line somehow. A quick search didn't turn up any obvious ways to break up a line in JSON though - any ideas @chrisgarrity?

Comment (redmangospros) [reply src/views/about/about.jsx]: Yeah.

File: src/views/about/about.jsx [hunk 4]
```diff
@@ -53,31 +53,62 @@ var About = React.createClass({
...
                                 allowFullScreen />
-                            <p><FormattedHTMLMessage id='about.literacyDescription' /></p>
+                            <p><FormattedMessage id='about.literacyDescription' /></p>
                         </li>
 
                         <li>
                             <h3><FormattedMessage id='about.aroundTheWorld' /></h3>
                             <img src="/images/about/around-the-world.png" alt="" />
-                            <p><FormattedHTMLMessage id='about.aroundTheWorldDescription' /></p>
+                            <p><FormattedMessage id='about.aroundTheWorldDescription' /></p>
```
Comment (chrisgarrity) [inline src/views/about/about.jsx]: Values is the second argument to `FormattedMessage`, so it needs to be inside the component tag. Getting the indentation/formatting will probably help with finding all of these; Lines 62-69 should be formatted as:
```jsx
<p>
    <FormattedMessage 
        id='about.aroundTheWorldDescription'
        values={{
            translationLink: (
                <a href='http://wiki.scratch.mit.edu/wiki/How_to_Translate_Scratch\'>
                    <FormattedMessage id='about.translationLinkText' />
                </a>
            )
        }}
    />
</p>

File: src/views/about/about.jsx (hunk 3 above)
Comment (chrisgarrity) [reply src/views/about/about.jsx]: you can add `// eslint-disable-next-line max-len` just before the long line to disable the check.

Comment (VutonDesign) [reply src/views/about/about.jsx]: Okay 👍

File: src/views/about/about.jsx (hunk 4 above)
Comment (VutonDesign) [reply src/views/about/about.jsx]: So I've just to make sure that everything is inside the ``<p>`` tag?

**Comment by VutonDesign** [comment]
@chrisgarrity I've managed to fix everything including the indentation errors but the maximum line length error is still there.

File: src/views/about/about.jsx [hunk 5]
```diff
@@ -102,8 +140,28 @@ var About = React.createClass({
 
                         <li>
                             <h3><FormattedMessage id='about.support' /></h3>
-                            <p><FormattedHTMLMessage id='about.supportDescription' /></p>
-                        </li>
+                            <p><FormattedMessage id='about.supportDescription'
+                            values={{
+ // eslint-disable-next-line max-len supportersList: 'National Science Foundation, Scratch Foundation, Siegel Family Endowment, Google, LEGO Foundation, Intel, Cartoon Network, Lemann Foundation, MacArthur Foundation',
```
Comment (VutonDesign) [inline src/views/about/about.jsx]: I've got a feeling I've done this wrong...

**Comment by VutonDesign** [comment]
I've added more commits to this pull by mistake and it's caused a lot of conflicts, So I've moved it over to #1714 

Sorry about this!

**Pull request closed without merging**